xdsl/interpreters/eqsat_pdl_interp.py

```diff
@@ -97,3 +113,56 @@ def run_getresults(
                     " that are used by a single EClassOp each."
                 )
         return (results,)
+
+    @impl(pdl_interp.GetDefiningOpOp)
+    def run_getdefiningop(
+        self,
+        interpreter: Interpreter,
+        op: pdl_interp.GetDefiningOpOp,
+        args: tuple[Any, ...],
+    ) -> tuple[Any, ...]:
+        defining_op = cast(
+            None | Operation,
+            PDLInterpFunctions.run_get_defining_op(self, interpreter, op, args).values[
+                0
+            ],
+        )
+        if isinstance(eclass_op := defining_op, eqsat.EClassOp):
```

Comment:
Let's reduce the indentation levels by bailing early instead?
```suggestion
        if not isinstance(defining_op, eqsat.EClassOp):
            return (defining_op,)
```

Reply:
This is more of a nit, feel free to merge as-is if you prefer

Reply:
I'd missed this one, should be OK now.